python/ray/workflow/tests/test_basic_workflows.py

```diff
@@ -323,6 +323,30 @@ def test_workflow_error_message():
     assert str(e.value) == expected_error_msg
 
 
+def test_options_update(workflow_start_regular_shared):
+    # Options are given in decorator first, then in the first .options()
+    # and finally in the second .options()
+    @workflow.step(name="old_name", metadata={"k": "v"}, max_retries=1, num_cpus=2)
+    def f():
+        return
+
+    new_f = f.options(name="new_name", metadata={"extra_k1": "extra_v1"}).options(
+        num_returns=2, metadata={"extra_k2": "extra_v2"}
+    )
```

Comment:
Can you add
```
    new_f = f.options(name="new_name", metadata={"extra_k1": "extra_v1"}).options(
        num_returns=2, metadata={"extra_k1": "extra_v2"}
    )
```
to make sure metadata overwriting also works here?

Reply:
@iycheng 
that's one behavior I wasn't sure - should metadata be like "update" (like ray_options) or "overwrite" (like max_retires).

I can see both has valid intuition and scenarios - 
1. update
I want to keep adding metadata during options-on-top-of-decorator or nested options. e.g.
```
@workflow.step(metadata = base_metadata)
def f()
f.options(metadata = some_metadata_I_add_during_runtime).run()

# sencarios:
@workflow.step(metadata = {"model_type": "model1"})
def f()
f.options(metadata = {"test_data": "data_v1"}).run(data_v1)
f.options(metadata = {"test_data": "data_v2"}).run(data_v2)
f.options(metadata = {"test_data": "data_v3"}).run(data_v3)
```
3. overwrite
I want to get rid of previous metadata
```
@workflow.step(metadata = metadata)
def f()
f.options(metadata = new_metadata).run()

# sencarios:
@workflow.step(metadata = {"model_type": "model1"})
def f()
# now I think the metadata in decorator isn't needed anymore or is wrong, and I want to replace/remove it.
f.options(metadata = {}).run()   # remove metadata

```

Reply:
The current behavior I did is to "update".  Thus in the test cases you see the result is the combination of metadata that is passed, whose logic is:  `decorator_metadata.update(options1_metadata).update(options2_metadata)`

Reply:
Ok. I think 1) update right now is more useful. Let's keep it for now. We can update it later if we need 2).